content/blog/experiences-with-ci-cd.pt-br.md

```diff
@@ -0,0 +1,476 @@
+num repositório "bare" hospedado na VPS que servia o website. O hook era nada
+mais do que um script shell, mais ou menos assim:
+
+```bash
+cd /path/to/the/worktree        # vá para um clone do repo
+git fetch origin                # obtenha as novas commits
+git reset --hard origin/master  # sincroniza com master
+make && make deploy             # faz o build e o deploy das páginas
+```
+
+Contanto que as permissões estivessem corretas e as dependências instaladas,
+tudo que eu precisava fazer para atualizar as páginas do seu site era editar os
+arquivos localmente, criar novas commits, e enviá-las para o remote. Se algum
+erro ocorresse, eu conseguiria vê-lo na saída do `git push`, pois este mostra a
+saída dos hooks enquanto são executados.
+
+[git-hooks]: https://git-scm.com/docs/githooks#post-receive
+[hugo]: https://gohugo.io
+
+### Por que não houve Integração Contínua
+
+Mesmo com toda minha felicidade na época com minha nova descoberta (hooks do
+Git) e minha configuração, em retrospecto não dá para dizer que havia CI. Antes
+de listar minhas razões, a seguir está a definição de Integração Contínua dada
+pela [Atlassian][atlassian-ci]:
+
+> Integração Contínua (CI) é a prática de automatizar a integração de mudanças
+> no código feita por múltiplos contribuidores em um só projeto de software. É
+> uma boa prática primária de DevOps, permitindo a desenvolvedores
+> frequentemente integrar mudanças a um repositório central, onde builds e
+> testes são então executados.
+
+Apesar dessa definição, eu pessoalmente acredito que pode-se atingir CI mesmo
+quando há somente um contribuidor, mas se você discorda de mim, adicione esse
+fato à seguinte lista de razões pelas quais não houve integração contínua no
+projeto em questão.
+
+**Razão #1:** Há somente um branch --- `master` ---, portanto mudanças sendo
+testadas já foram integradas.
+
+O propósito de CI é testar e revisar mudanças antes de integrá-las ao branch de
+destino. Suítes como GitHub, GitLab, Bitbucket, entre outras são extremamente
+úteis: elas materializam a ideia de proposta de mudança em pull (ou merge)
+requests. Estas requests podem ser negadas, aceitas, revisadas, etc., e ainda
+mais legal é o fato de que quando são criadas ou modificadas, a suíte pode nos
+notificar programaticamente por meio de webhooks, e até tratar dos eventos
+nativamente.
+
+No meu caso, não haviam pull requests: commits eram adicionadas à `master` e
+empurradas __diretamente upstream,__ sem mais nem menos.
+
+**Razão #2:** O único teste feito é deployment.
+
+Não há teste automatizado de nenhum tipo, nem mesmo para ver se as páginas ainda
+seriam geradas. Consigo até lembrar de momentos em que quebrei a geração das
+páginas enquanto testava as capacidades de template do Hugo --- a versão
+instalada na minha máquina era diferente da instalada no servidor.
+
+A falta de um ambiente facilmento reproduzido para testes (e obviamente os
```

Comment:
"facilmente"

Reply:
Sorry for the typos. Sometimes I skip a few words as I transfer the phrase from my head to the screen, so hopefully there haven't been many instances of that. Fixed in ead6a2d.